File: test_regress/t/t_multitop1.pl
#!/usr/bin/env perl
if (!$::Driver) { use FindBin; exec("$FindBin::Bin/bootstrap.pl", @ARGV, $0); die; }
# DESCRIPTION: Verilator: Verilog Test driver/expect definition
#
# Copyright 2003-2009 by Wilson Snyder. This program is free software; you
# can redistribute it and/or modify it under the terms of either the GNU
# Lesser General Public License Version 3 or the Perl Artistic License
# Version 2.0.
# SPDX-License-Identifier: LGPL-3.0-only OR Artistic-2.0

scenarios(vlt_all => 1);

compile(
    );

execute(
    check_finished => 0,
    );

# Order of lines is unspecified, so don't use a golden file
file_grep($Self->{run_log_filename}, qr!In 'top.t'!);
file_grep($Self->{run_log_filename}, qr!In 'top.t.s'!);
file_grep_not($Self->{run_log_filename}, qr!in_subfile!);

ok(1);
1;

File: test_regress/t/t_multitop1.py
#!/usr/bin/env python3
# DESCRIPTION: Verilator: Verilog Test driver/expect definition
#
# Copyright 2024 by Wilson Snyder. This program is free software; you
# can redistribute it and/or modify it under the terms of either the GNU
# Lesser General Public License Version 3 or the Perl Artistic License
# Version 2.0.
# SPDX-License-Identifier: LGPL-3.0-only OR Artistic-2.0

import vltest_bootstrap

test.scenarios('vlt_all')

test.compile()

test.execute(check_finished=False)

# Order of lines is unspecified, so don't use a golden file
test.file_grep(test.run_log_filename, r"In 'top.t'")
test.file_grep(test.run_log_filename, r"In 'top.t.s'")
test.file_grep_not(test.run_log_filename, r"in_subfile")

test.passes()
